File: src/main/java/net/p35/commonenumvalues/macro/DevelopersMacro.java
package net.p35.commonenumvalues.macro;

import com.intellij.codeInsight.lookup.LookupElement;
import com.intellij.codeInsight.lookup.LookupElementBuilder;
import com.intellij.codeInsight.template.Expression;
import com.intellij.codeInsight.template.ExpressionContext;
import com.intellij.codeInsight.template.macro.EnumMacro;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;

public class DevelopersMacro extends EnumMacro {
    @Override
    public String getName() {
        return "developers";
    }

    @Override
    public String getPresentableName() {
        return "developers";
    }

    @Nullable
    @Override
    public LookupElement[] calculateLookupItems(@NotNull Expression[] params, ExpressionContext context) {
        ArrayList<LookupElementBuilder> developers = new ArrayList<>();
        String[] devNames = new String[]{"pavlo", "oksana", "dmytro", "dylan", "david"};
        for (String devName : devNames) {
            developers.add(LookupElementBuilder.create(devName));
        }

        return developers.toArray(new LookupElement[0]);
    }
}

File: src/main/java/net/p35/commonenumvalues/macro/DevelopersMacro.kt
package net.p35.commonenumvalues.macro

import com.intellij.codeInsight.lookup.LookupElement
import com.intellij.codeInsight.lookup.LookupElementBuilder
import com.intellij.codeInsight.template.Expression
import com.intellij.codeInsight.template.ExpressionContext
import com.intellij.codeInsight.template.macro.EnumMacro

import java.util.ArrayList

class DevelopersMacro : EnumMacro() {
    override fun getName(): String {
        return "developers"
    }

    override fun getPresentableName(): String {
        return "developers"
    }

    override fun calculateLookupItems(params: Array<Expression>, context: ExpressionContext?): Array<LookupElement>? {
        val developers = ArrayList<LookupElementBuilder>()
        val devNames = arrayOf("pavlo", "oksana", "dmytro", "dylan", "david")
        for (devName in devNames) {
            developers.add(LookupElementBuilder.create(devName))
        }

        return developers.toTypedArray()
    }
}